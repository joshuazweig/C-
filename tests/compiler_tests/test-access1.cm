int main()
{

	stone a;
	stone b;
	stone c;
	stone d;
	mint m; 
	a = "3";
	b = "5";
	m = <a, b>;

	printf("Calling print_stone on a (=3): ");
	print_stone(a);

	printf("Calling print_stone on b (=5): ");
	print_stone(b);

	c = access_mint(m, 0);
	printf("Printing result of access_mint(m, 0), should = a: ");
	print_stone(c);

	d = access_mint(m, 1);
	printf("Printing result of access_mint(m, 1), should = b: ");
	print_stone(d);
		
	return 0;
}
