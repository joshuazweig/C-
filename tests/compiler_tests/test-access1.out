Calling print_stone on a (=3): 3
Calling print_stone on b (=5): 5
Printing result of access_mint(m, 0), should = a: 3
Printing result of access_mint(m, 1), should = b: 5